OpenSim/Framework/Servers/BaseOpenSimServer.cs

```diff
@@ -311,12 +311,21 @@ public virtual void Startup()
         /// </summary>      
         public virtual void Shutdown()
         {
+            Shutdown(0);
```

Comment:
Technically both restart and shutdown are clean exits so yes, I agree the proper return value would be a 0.  However, if a 0 was returned as in the past, it be ignored by all callers, and not allow the admin to use the ```shutdown``` console command without aggravation. The whole purpose of this PR is to allow the ```shutdown``` console command to be used rather than an external system.  In both cases, the region is notified to shutdown by a human operator.  The region is not making any decisions, it is the recipient of the command from the human operator.

This PR attempts to allow any callers to _optionally_ recognize the _abnormal_ case of a manual intervention by the human, requesting an exit that should **not** be turned into a restart like it normally would.  That exceptional case is the direct result of use of the shutdown command on the console.  All other terminations, graceful or crashes, would result in a restart.

Because Halcyon _supports_ the ```shutdown``` console command (at all), I don't have a problem with differentiating that on the exit, if it allows the _operator's_ intentions to be recognized. If we don't, then the alternative, really, is to remove the shutdown command, since we're effectively saying don't use it for a shutdown where you _don't_ want a restart. I don't think that is practical and I don' think it's a philosophical crime to _allow_ the operator to manually intervene. However, I do think it would be a philosophical crime to _require_ the operator to intervene.

And that is why I don't like the exceptional case of a shutdown (no restart) returning a standard exit code, especially the most common 0 case. It is an exceptional request of manual intervention by an operator to bring a region down and intend for it to stay down, rather than to allow any controlling software to restart it automatically (including a batch script in a loop).  It is an exceptional case and we don't need to pick the _most normal_ of exit codes.

However, the last two sentences above aren't correct; no exit code requires a caller to check it. For example, neither Zookeeper nor Maestro would check it.  It only _allows_ the grid owner to differentiate if convenient to their process, as is the case on GW grid, which is what triggered this PR.